src/librawspeed/decoders/ArwDecoder.cpp

```diff
@@ -118,8 +119,9 @@ RawImage ArwDecoder::decodeRawInternal() {
       // And now decode as a normal 16bit raw
       mRaw->dim = iPoint2D(width, height);
       mRaw->createData();
-      ByteStream input(mFile, off, len);
-      Decode16BitRawBEunpacked(input, width, height);
+
+      UncompressedDecompressor u(*mFile, off, len, mRaw, uncorrectedRawValues);
+      u.Decode16BitRawBEunpacked(width, height);
```

Comment:
Please consider changing the naming of the different `Decode...` methods to use a lowercase `decode...` as this is the naming style used almost universally in the rest of the library. And since you just touched each of those functions anyway, seems like the right time.

Also, simply because I personally think `UncompressedDecompressor` sounds bit 'contradicting': maybe something like `PlainDecompressor` or `UnpackDecompressor` would fit better?